dali/pipeline/operators/crop/bbox_crop.h

```diff
@@ -88,24 +106,22 @@ class RandomBBoxCrop : public Operator<Backend> {
 
   void WriteLabelsToOutput(SampleWorkspace *ws, const std::vector<int> &labels);
 
-  float SelectMinimumOverlap() {
-    static std::uniform_int_distribution<> sampler(
-        0, static_cast<int>(thresholds_.size() - 1));
-    return thresholds_[sampler(rd_)];
+  std::pair<float, bool> SelectMinimumOverlap() {
+    std::uniform_int_distribution<> sampler(
+      0, static_cast<int>(sample_options_.size() - 1));
+    return sample_options_[sampler(rd_)];
   }
 
   float SampleCandidateDimension() {
-    static std::uniform_real_distribution<> sampler(scaling_bounds_.min,
-                                                    scaling_bounds_.max);
+    std::uniform_real_distribution<> sampler(scaling_bounds_.min, scaling_bounds_.max);
     return static_cast<float>(sampler(rd_));
   }
 
   bool ValidAspectRatio(float width, float height) const {
     return aspect_ratio_bounds_.Contains(width / height);
   }
 
-  bool ValidOverlap(const Crop &crop, const BoundingBoxes &boxes,
-                    float threshold) {
+  bool ValidOverlap(const Crop &crop, const BoundingBoxes &boxes, float threshold) {
```

Comment:
A few functions here could be const qualified member functions

Reply:
Done